lensless/hardware/mask.py

```diff
@@ -0,0 +1,398 @@
+            default value: MLS
+        n_bits: int, optional
+            Number of bits for row/column for pattern generation.
+            size = `4*n_bits + 1` for MURA and `2^n - 1` for MLS.
+            Default is 8 (for a 255x255 MLS mask).
+        """
+
+        self.row = None
+        self.column = None
+        self.method = method
+        self.n_bits = n_bits
+
+        super().__init__(**kwargs)
+
+    def create_mask(self):
+        """
+        Creating coded aperture mask using either the MURA of MLS method.
+        """
+        assert self.method in ["MURA", "MLS"], "Method should be either 'MLS' or 'MURA'"
+
+        # Generating pattern
+        if self.method == "MURA":
+            self.mask = self.squarepattern(4 * self.n_bits + 1)[1:, 1:]
+            self.row = 2 * self.mask[0, :] - 1
+            self.column = 2 * self.mask[:, 0] - 1
+        else:
+            seq = max_len_seq(self.n_bits)[0] * 2 - 1
+            h_r = np.r_[seq, seq]
+            self.row = h_r
+            self.column = h_r
+            self.mask = (np.outer(h_r, h_r) + 1) / 2
+
+            # Upscaling
+        if np.any(self.sensor_resolution != self.mask.shape):
+            upscale_factor_height = self.sensor_resolution[0] / self.mask.shape[0]
+            upscale_factor_width = self.sensor_resolution[1] / self.mask.shape[1]
+            upscaled_mask = zoom(self.mask, (upscale_factor_height, upscale_factor_width))
+            upscaled_mask = np.clip(upscaled_mask, 0, 1)
+            self.mask = np.round(upscaled_mask).astype(int)
+
+    def is_prime(self, n):
+        """
+        Assess whether a number is prime or not.
+
+        Parameters
+        ----------
+        n: int
+            The number we want to check.
+        """
+        if n % 2 == 0 and n > 2:
+            return False
+        return all(n % i for i in range(3, int(sqrt(n)) + 1, 2))
+
+    def squarepattern(self, p):
+        """
+        Generate MURA square pattern.
+
+        Parameters
+        ----------
+        p: int
+            number of bits.
+        """
+        if not self.is_prime(p):
+            raise ValueError("p is not a valid length. It must be prime.")
+        A = np.zeros((p, p), dtype=int)
+        q = quadratic_residues(p)
+        A[1:, 0] = 1
+        for j in range(1, p):
+            for i in range(1, p):
+                if not ((i - 1 in q) != (j - 1 in q)):
+                    A[i, j] = 1
+        return A
+
+
+class PhaseContour(Mask):
+    """
+    Phase contour subclass of the Mask class
+    From the PhlatCam article https://ieeexplore.ieee.org/document/9076617
+    """
+
+    def __init__(
+        self, noise_period=(8, 8), refractive_index=1.2, n_iter=10, design_wv=532e-9, **kwargs
+    ):
+        """
+        Phase contour mask contructor (PhlatCam).
+
+        Parameters
+        ----------
+        noise_period: tuple (dim=2)
+            noise period of the Perlin noise (px)
+            default value: (8,8)
```

Comment:
Same line as above. "Default value is (8, 8)."
Remember to follow good punctuation!